TicTacToe/TicTacToe.cs

```diff
@@ -32,46 +33,86 @@ public static void GetInput()
             int row = int.Parse(Console.ReadLine());
             Console.WriteLine("Enter Column:");
             int column = int.Parse(Console.ReadLine());
+
+            PlaceMark(row, column);
+
+            // playerTurn == "X" ? playerTurn = "O" : playerTurn = "X";
         }
 
-        public static void PlaceMark(int row, int column)
+        public static bool PlaceMark(int row, int column)
         {
-        // your code goes here
-        }
+            if (board[row][column] != " ")
+            {
+                Console.WriteLine("The row column is already taken");
+                return false;
+            }
+
+            board[row][column] = playerTurn;
 
+            return true;
+        }
         public static bool CheckForWin()
         {
-            // your code goes here
+            if (HorizontalWin())
+            {
+                return true;
+            }
+            else if (VerticalWin())
+            {
+                return true;
+            }
+            else if (DiagonalWin())
+            {
+                return true;
+            }
+
 
             return false;
+
         }
 
+
         public static bool CheckForTie()
         {
             // your code goes here
 
             return false;
         }
-        
+
         public static bool HorizontalWin()
         {
-        // your code goes here
-
-        return false;
+            for (int i = 0; i < board.GetLength(0); i++)
+            {
+                for (int j = 0; j < board.GetLength(0) - 1; j++)
+                {
+                    if (board[i][j] != board[i][j + 1])
```

Comment:
What if the cells of the first row are equal to each other, and they are blank? is that a win?